Payload limit and configuration guide (#385)
meilisearch/documentation: guides/advanced_guides/configuration.md

# Configuration

Many options are available to configure a MeiliSearch Instance. Each of these options are added on MeiliSearch Instance launch.

Options can be either communicated through **environment variables** or **command line options**.

## Passing arguments via the command line

Options are added on launch.

```bash
$ ./meilisearch --db-path ./meilifiles --http-addr '127.0.0.1:7700'
Server is listening on: http://127.0.0.1:7700
```

## Passing arguments via the environment variables

> The format of the environment variables are the same as the command line options with the exception that it is uppercased and `MEILI_` is added at the start.

Reviewer: ```suggestion
The format of the environment variables is identical to the command line options with the exception that it is uppercased and `MEILI_` is added at the beginning.
```

```
$ export MEILI_DB_PATH=./meilifiles
$ export MEILI_HTTP_ADDR=127.0.0.1:7700
$ ./meilisearch
Server is listening on: http://127.0.0.1:7700
```

## Options

List of options:

- [Path to database](/guides/advanced_guides/configuration.md#path-to-database)
- [HTTP address](/guides/advanced_guides/configuration.md#http-address)
- [Master Key](/guides/advanced_guides/configuration.md#master-key)
- [No Analytics](/guides/advanced_guides/configuration.md#no-analytics)
- [Environment](/guides/advanced_guides/configuration.md#environment)
- [Payload size limit](/guides/advanced_guides/configuration.md#payload-size-limit)

### Path to database

**Environment Variable**: `MEILI_DB_PATH`
**CLI option**: `--db-path`

Defines the location for the database files

**Default value**: `"./data.ms"`

### HTTP Address

**Environment Variable**: `MEILI_HTTP_ADDR`
**CLI option**: `--http-addr`

> he address on which the http server will listen.

Reviewer: ```suggestion
The address the HTTP server will listen on.
```

**Default value**: `"127.0.0.1:7700"`

### Master Key

**Environment Variable**: `MEILI_MASTER_KEY`
**CLI option**: `--master-key`

> The master key allowing you to do everything on the server. If no master key is provided

Reviewer: The last sentence seems like unfinished. Is it normal?
